### PR TITLE
[feat] add support for cudagraph and its unit test.

### PR DESCRIPTION
**Pull request by liutongxuan**

### CODE REVIEW

File: src/engine/worker.cpp [hunk 1]
```diff
@@ -18,8 +19,59 @@
...
+    input_buffer_.emplace("flatten_positions", flatten_positions);
+    // input_buffer_.emplace("kv_cache", kv_cache);
+    // TODO: slot_mapping, context_lens, block_tables
+    output_buffer_.emplace("logits", logits);
+  }
+
+  // ids, positions, kv_cache, input_meta
+  torch::Tensor forward(torch::Tensor flatten_tokens, torch::Tensor flatten_positions,
+      const InputParameters& params, std::vector<KVCache>& kv_cache) {
+    input_buffer_["flatten_tokens"] = flatten_tokens,
```
Comment (guocuimi) [inline src/engine/worker.cpp:62]: i think you need to copy the value from input faltten_tokens to input_buffer, otherwise, replay would still use the old tensor instead of passed in tensor.

File: src/engine/worker.cpp [hunk 2]
```diff
@@ -18,8 +19,59 @@
...
+    model_->forward(flatten_tokens,
+                    flatten_positions,
+                    kv_cache,
+                    params);
+    torch::cuda::synchronize();
+
+    // create cudagraph and capture
+    graph_ = std::make_unique<at::cuda::CUDAGraph>();
+    graph_->capture_begin();
+    auto logits = model_->forward(flatten_tokens,
```
Comment (guocuimi) [inline src/engine/worker.cpp:44]: the logits is var len, depends on the number of seqs in the batch. not sure if graph support this.

File: src/engine/worker.cpp [hunk 3]
```diff
@@ -18,8 +19,59 @@
 #include "sampling/logits_processor.h"
 #include "sampling/sampler.h"
 
+DEFINE_bool(enable_cudagraph, false, "Enable CUDAGraph to optimize model execution.");
+
 namespace llm {
 
+class CudaGraphRunner {
```
Comment (guocuimi) [inline src/engine/worker.cpp:26]: sounds we can create a simple model to test this cuda graph runner.

**Comment by guocuimi** [comment]
One important aspect to note is that we're currently utilizing different kernels for prefill and decode. As a result, this leads to the capture of two distinct graphs for each operation. For now, let's prioritize enabling graph capture for decode initially. what do you think?

File: src/engine/worker.cpp (hunk 3 above)
Comment (liutongxuan) [reply src/engine/worker.cpp:36]: yep
